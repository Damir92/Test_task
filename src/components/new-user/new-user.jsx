import React, {useState} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {Button, Dialog, DialogTitle, DialogContent, TextField, makeStyles} from '@material-ui/core';

import {postNewUser} from '../../store/api-actions';

const useStyles = makeStyles(() => ({
  form: {
    display: `grid`,
    gridTemplateColumns: `1fr 1fr`,
    gap: `20px`,
  },
}));

const NewUser = ({onSubmit}) => {
  const classes = useStyles();

  const [showModal, setShowModal] = useState(false);

  const handleClickBtn = () => {
    setShowModal(true);
  };

  const handleClose = () => {
    setShowModal(false);
  };

  const handleSubmit = (evt) => {
    evt.preventDefault();
    onSubmit({
      firstName: evt.target.firstName.value,
      lastName: evt.target.lastName.value,
      group: evt.target.group.value,
    });
    setShowModal(false);
  };

  return (
    <>
      <Button
        variant='outlined'
        color='primary'
        onClick={handleClickBtn}
      >
        Add new user
      </Button>
      <Dialog
        open={showModal}
        onClose={handleClose}
      >
        <DialogTitle>
          Add data for new user
        </DialogTitle>
        <DialogContent>
          <form
            className={classes.form}
            onSubmit={handleSubmit}
          >
            <TextField
              required
              label="First name"
              name="firstName"
            />
            <TextField
              required
              label="Last name"
              name="lastName"
            />
            <TextField
              label="Group"
              name="group"
            />
            <Button
              type="submit"
              variant="outlined"
            >
              Send
            </Button>
          </form>
        </DialogContent>
      </Dialog>
    </>
  );
};

NewUser.propTypes = {
  onSubmit: PropTypes.func.isRequired,
};

const mapDispatchToProps = (dispatch) => ({
  onSubmit: (data) => dispatch(postNewUser(data)),
});

export {NewUser};
export default connect(null, mapDispatchToProps)(NewUser);
