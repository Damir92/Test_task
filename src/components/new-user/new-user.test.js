import React from 'react';
import {shallow} from 'enzyme';

import {NewUser} from './new-user';

describe(`New user component`, () => {
  it(`NewUser snapshot`, () => {
    const tree = shallow(
        <NewUser
          onSubmit={() => {}}
        />
    );

    expect(tree).toMatchSnapshot();
  });
});
