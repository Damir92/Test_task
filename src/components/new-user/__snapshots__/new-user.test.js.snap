// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New user component NewUser snapshot 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Button))
    color="primary"
    onClick={[Function]}
    variant="outlined"
  >
    Add new user
  </WithStyles(ForwardRef(Button))>
  <WithStyles(ForwardRef(Dialog))
    onClose={[Function]}
    open={false}
  >
    <WithStyles(ForwardRef(DialogTitle))>
      Add data for new user
    </WithStyles(ForwardRef(DialogTitle))>
    <WithStyles(ForwardRef(DialogContent))>
      <form
        className="makeStyles-form-1"
        onSubmit={[Function]}
      >
        <WithStyles(ForwardRef(TextField))
          label="First name"
          name="firstName"
          required={true}
        />
        <WithStyles(ForwardRef(TextField))
          label="Last name"
          name="lastName"
          required={true}
        />
        <WithStyles(ForwardRef(TextField))
          label="Group"
          name="group"
        />
        <WithStyles(ForwardRef(Button))
          type="submit"
          variant="outlined"
        >
          Send
        </WithStyles(ForwardRef(Button))>
      </form>
    </WithStyles(ForwardRef(DialogContent))>
  </WithStyles(ForwardRef(Dialog))>
</Fragment>
`;
